src/modules/submissions.js

```diff
@@ -2604,44 +2607,67 @@
 
 						shouldTeahouse.resolve( !hasTeahouseCat );
 					} );
-				}
 
-				$.when( shouldTeahouse ).then( ( teahouse ) => {
-					let message;
-					if ( isDecline ) {
-						message = AFCH.msg.get( 'declined-submission', {
-							$1: AFCH.consts.pagename,
-							$2: afchSubmission.shortTitle,
-							$3: ( declineReason === 'cv' || declineReason2 === 'cv' ) ?
-								'yes' : 'no',
-							$4: declineReason,
-							$5: newParams[ '3' ] || '',
-							$6: declineReason2 || '',
-							$7: newParams.details2 || '',
-							$8: ( declineReason === 'reason' || declineReason2 === 'reason' ) ?
-								'' : data.declineTextarea
-						} );
-					} else {
-						message = AFCH.msg.get( 'rejected-submission', {
-							$1: AFCH.consts.pagename,
-							$2: afchSubmission.shortTitle,
-							$3: data.rejectReason[ 0 ],
-							$4: '',
-							$5: data.rejectReason[ 1 ] || '',
-							$6: '',
-							$7: data.rejectTextarea
+					// Check if the user has already been welcomed.
+					if ( data.sendWelcome ) {
+						userTalk.getCategories( /* useApi */ true ).done( ( categories ) => {
+							let hasWelcomeDraftCat = false;
+							const welcomeDraftCategories = [
+								'Category:Wikipedians who have received an AfC welcome message'
+							];
+
+							$.each( categories, ( _, cat ) => {
+								if ( welcomeDraftCategories.indexOf( cat ) !== -1 ) {
+									hasWelcomeDraftCat = true;
+									return false;
+								}
+							} );
+
+							shouldWelcomeUser.resolve( !hasWelcomeDraftCat );
 						} );
 					}
 
-					if ( teahouse ) {
-						message += '\n\n' + AFCH.msg.get( 'teahouse-invite' );
-					}
+					$.when( shouldTeahouse, shouldWelcomeUser ).then( ( teahouse, welcome ) => {
```

Comment:
I think I found a bug.

<img width="422" alt="image" src="https://github.com/user-attachments/assets/085378aa-490b-4bfe-a9f8-64d601f8bccb">

On the master branch, if I do this, the subumitter gets the decline message on their user talk.

With this patch loaded, if I do this, the submitter gets zero messages posted to their user talk.

Can you fix when you get a chance?

Reply:
I will fix it this weekend.